// @dicebear/pixel-art/dist/index.es.js
import { utils } from '@dicebear/avatars';

const schema$1 = {
  "title": "Options",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "properties": {
    "beard": {
      "title": "Beard",
      "type": "array",
      "items": {
        "type": "string",
        "enum": ["variant04", "variant03", "variant02", "variant01"]
      },
      "default": ["variant04", "variant03", "variant02", "variant01"]
    },
    "beardProbability": {
      "title": "Beard Probability",
      "default": 10,
      "type": "integer",
      "minimum": 0,
      "maximum": 100
    },
    "eyes": {
      "title": "Eyes",
      "type": "array",
      "items": {
        "type": "string",
        "enum": ["variant13", "variant12", "variant11", "variant10", "variant09", "variant08", "variant07", "variant06", "variant05", "variant04", "variant03", "variant02", "variant01"]
      },
      "default": ["variant13", "variant12", "variant11", "variant10", "variant09", "variant08", "variant07", "variant06", "variant05", "variant04", "variant03", "variant02", "variant01"]
    },
    "eyebrows": {
      "title": "Eyebrows",
      "type": "array",
      "items": {
        "type": "string",
        "enum": ["variant13", "variant12", "variant11", "variant10", "variant09", "variant08", "variant07", "variant06", "variant05", "variant04", "variant03", "variant02", "variant01"]
      },
      "default": ["variant13", "variant12", "variant11", "variant10", "variant09", "variant08", "variant07", "variant06", "variant05", "variant04", "variant03", "variant02", "variant01"]
    },
    "mouth": {
      "title": "Mouth",
      "type": "array",
      "items": {
        "type": "string",
        "enum": ["surprised03", "surprised02", "happy09", "happy08", "happy07", "happy06", "happy05", "happy04", "happy03", "happy02", "happy01", "sad08", "sad07", "sad06", "sad05", "sad04", "sad03", "sad02", "sad01", "surprised01"]
      },
      "default": ["surprised03", "surprised02", "happy09", "happy08", "happy07", "happy06", "happy05", "happy04", "happy03", "happy02", "happy01", "sad08", "sad07", "sad06", "sad05", "sad04", "sad03", "sad02", "sad01", "surprised01"]
    },
    "hair": {
      "title": "Hair",
      "type": "array",
      "items": {
        "type": "string",
        "enum": ["short11", "short10", "short09", "short08", "short07", "short06", "short05", "short04", "short03", "short02", "long15", "short01", "long14", "long13", "long12", "long11", "long10", "long09", "long08", "long07", "long06", "long05", "long04", "long03", "long02", "long01"]
      },
      "default": ["short11", "short10", "short09", "short08", "short07", "short06", "short05", "short04", "short03", "short02", "long15", "short01", "long14", "long13", "long12", "long11", "long10", "long09", "long08", "long07", "long06", "long05", "long04", "long03", "long02", "long01"]
    },
    "hairProbability": {
      "title": "Hair Probability",
      "default": 100,
      "type": "integer",
      "minimum": 0,
      "maximum": 100
    },
    "accessories": {
      "title": "Accessories",
      "type": "array",
      "items": {
        "type": "string",
        "enum": ["variant04", "variant03", "variant02", "variant01"]
      },
      "default": ["variant04", "variant03", "variant02", "variant01"]
    },
    "accessoriesProbability": {
      "title": "Accessories Probability",
      "default": 10,
      "type": "integer",
      "minimum": 0,
      "maximum": 100
    },
    "glasses": {
      "title": "Glasses",
      "type": "array",
      "items": {
        "type": "string",
        "enum": ["variant07", "variant06", "variant05", "variant04", "variant03", "variant02", "variant01"]
      },
      "default": ["variant07", "variant06", "variant05", "variant04", "variant03", "variant02", "variant01"]
    },
    "glassesProbability": {
      "title": "Glasses Probability",
      "default": 20,
      "type": "integer",
      "minimum": 0,
      "maximum": 100
    },
    "hat": {
      "title": "Hat",
      "type": "array",
      "items": {
        "type": "string",
        "enum": ["variant01", "variant02", "variant03", "variant04", "variant05", "variant06", "variant07", "variant08", "variant09", "variant10", "variant11", "variant12"]
      },
      "default": ["variant01", "variant02", "variant03", "variant04", "variant05", "variant06", "variant07", "variant08", "variant09", "variant10", "variant11", "variant12"]
    },
    "hatProbability": {
      "title": "Hat Probability",
      "default": 10,
      "type": "integer",
      "minimum": 0,
      "maximum": 100
    },
    "clothing": {
      "title": "Clothing",
      "type": "array",
      "items": {
        "type": "string",
        "enum": ["variant25", "variant24", "variant23", "variant22", "variant21", "variant20", "variant19", "variant18", "variant17", "variant16", "variant15", "variant14", "variant13", "variant12", "variant11", "variant10", "variant09", "variant08", "variant07", "variant06", "variant05", "variant04", "variant03", "variant02", "variant01"]
      },
      "default": ["variant25", "variant24", "variant23", "variant22", "variant21", "variant20", "variant19", "variant18", "variant17", "variant16", "variant15", "variant14", "variant13", "variant12", "variant11", "variant10", "variant09", "variant08", "variant07", "variant06", "variant05", "variant04", "variant03", "variant02", "variant01"]
    },
    "skinColor": {
      "title": "Skin Color",
      "type": "array",
      "items": {
        "anyOf": [{
          "type": "string",
          "enum": ["variant01", "variant02", "variant03", "variant04", "variant05", "variant06", "variant07", "variant08"]
        }, {
          "type": "string",
          "pattern": "^#([a-fA-F0-9]{3}|[a-fA-F0-9]{4}|[a-fA-F0-9]{6}|[a-fA-F0-9]{8})$"
        }]
      },
      "default": ["variant01", "variant02", "variant03", "variant04", "variant05", "variant06", "variant07", "variant08"]
    },
    "hairColor": {
      "title": "Hair Color",
      "type": "array",
      "items": {
        "anyOf": [{
          "type": "string",
          "enum": ["variant01", "variant02", "variant03", "variant04", "variant05", "variant06", "variant07", "variant08", "variant09", "variant10"]
        }, {
          "type": "string",
          "pattern": "^#([a-fA-F0-9]{3}|[a-fA-F0-9]{4}|[a-fA-F0-9]{6}|[a-fA-F0-9]{8})$"
        }]
      },
      "default": ["variant01", "variant02", "variant03", "variant04", "variant05", "variant06", "variant07", "variant08", "variant09", "variant10"]
    },
    "accessoriesColor": {
      "title": "Accessories Color",
      "type": "array",
      "items": {
        "anyOf": [{
          "type": "string",
          "enum": ["gold01", "gold02", "gold03", "silver01", "silver02"]
        }, {
          "type": "string",
          "pattern": "^#([a-fA-F0-9]{3}|[a-fA-F0-9]{4}|[a-fA-F0-9]{6}|[a-fA-F0-9]{8})$"
        }]
      },
      "default": ["gold01", "gold02", "gold03", "silver01", "silver02"]
    },
    "mouthColor": {
      "title": "Mouth Color",
      "type": "array",
      "items": {
        "anyOf": [{
          "type": "string",
          "enum": ["variant01", "variant02", "variant03", "variant04"]
        }, {
          "type": "string",
          "pattern": "^#([a-fA-F0-9]{3}|[a-fA-F0-9]{4}|[a-fA-F0-9]{6}|[a-fA-F0-9]{8})$"
        }]
      },
      "default": ["variant01", "variant02", "variant03", "variant04"]
    },
    "clothesColor": {
      "title": "Clothes Color",
      "type": "array",
      "items": {
        "anyOf": [{
          "type": "string",
          "enum": ["yellow01", "yellow02", "yellow03", "red01", "red02", "red03", "blue01", "blue02", "blue03", "green01", "green02", "green03"]
        }, {
          "type": "string",
          "pattern": "^#([a-fA-F0-9]{3}|[a-fA-F0-9]{4}|[a-fA-F0-9]{6}|[a-fA-F0-9]{8})$"
        }]
      },
      "default": ["yellow01", "yellow02", "yellow03", "red01", "red02", "red03", "blue01", "blue02", "blue03", "green01", "green02", "green03"]
    },
    "hatColor": {
      "title": "Hat Color",
      "type": "array",
      "items": {
        "anyOf": [{
          "type": "string",
          "enum": ["pink01", "blue01", "green01", "purple01", "red01", "black01", "gray01"]
        }, {
          "type": "string",
          "pattern": "^#([a-fA-F0-9]{3}|[a-fA-F0-9]{4}|[a-fA-F0-9]{6}|[a-fA-F0-9]{8})$"
        }]
      },
      "default": ["pink01", "blue01", "green01", "purple01", "red01", "black01", "gray01"]
    },
    "glassesColor": {
      "title": "Glasses Color",
      "type": "array",
      "items": {
        "anyOf": [{
          "type": "string",
          "enum": ["green01", "blue01", "red01", "black01", "black02", "black03"]
        }, {
          "type": "string",
          "pattern": "^#([a-fA-F0-9]{3}|[a-fA-F0-9]{4}|[a-fA-F0-9]{6}|[a-fA-F0-9]{8})$"
        }]
      },
      "default": ["green01", "blue01", "red01", "black01", "black02", "black03"]
    }
  },
  "additionalProperties": false
};

const beard = {
  'variant04': (components, colors) => "\n<path opacity=\".9\" d=\"M4 10v3h1v1h1v1h8v-1h1v-1h1v-3h-2v1H6v-1H4Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'variant03': (components, colors) => "\n<path opacity=\".9\" d=\"M4 13h1v1h1v1h8v-1h1v-1h1v-3h-1v1h-1v1H6v-1H5v-1H4v3Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'variant02': (components, colors) => "\n<path opacity=\".9\" d=\"M4 11v2h1v1h1v1h8v-1h1v-1h1v-2H4Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'variant01': (components, colors) => "\n<path opacity=\".9\" d=\"M4 8v5h1v1h1v1h8v-1h1v-1h1V8h-1v2h-1v1h-1l-.059 1H7v-1H6v-1H5V8H4Z\" fill=\"".concat(colors.hair.value, "\"/>\n")
};

const eyes = {
  'variant13': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M5 9V7h3v2H5Zm7-2h3v2h-3V7Z\" fill=\"#fff\"/>\n<path d=\"M7 8v1h1V8H7Zm7 0h1v1h-1V8Z\" fill=\"#000\"/>\n</g>\n",
  'variant12': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M5 7h3v2H5V7Zm7 0h3v2h-3V7Z\" fill=\"#fff\"/>\n<path d=\"M6 8h1v1H6V8Zm7 1V8h1v1h-1Z\" fill=\"#000\"/>\n</g>\n",
  'variant11': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M5 7h3v2H5V7Zm7 0h3v2h-3V7Z\" fill=\"#fff\"/>\n<path d=\"M7 8h1v1H7V8Zm5 0h1v1h-1V8Z\" fill=\"#000\"/>\n</g>\n",
  'variant10': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M6 7h1v1h1v1H6V7Zm6 0h1v1h1v1h-2V7Z\" fill=\"#fff\"/>\n<path d=\"M6 8h1v1H6V8Zm6 0h1v1h-1V8Z\" fill=\"#000\"/>\n</g>\n",
  'variant09': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M5 8h2v1H5V8Zm7 0h2v1h-2V8Z\" fill=\"#fff\"/>\n<path d=\"M7 8h1v1H7V8Zm7 0h1v1h-1V8Z\" fill=\"#000\"/>\n</g>\n",
  'variant08': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M6 8h1v1H6V8Zm7 0h1v1h-1V8Z\" fill=\"#fff\"/>\n<path d=\"M7 8h1v1H7V8Zm5 0h1v1h-1V8Z\" fill=\"#000\"/>\n</g>\n",
  'variant07': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M5 7v1h3V7H5Zm7 0h3v1h-3V7Z\" fill=\"#fff\"/>\n<path d=\"M5 9V8h1V7h1v1h1v1H5Zm7 0V8h1V7h1v1h1v1h-3Z\" fill=\"#000\"/>\n<path d=\"M5 9V8h1V7h1v1h1v1H7V8H6v1H5Zm7 0V8h1V7h1v1h1v1h-1V8h-1v1h-1Z\" fill=\"#fff\" fill-opacity=\".5\"/>\n</g>\n",
  'variant06': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M5 8h3v1H5V8Zm7 0h3v1h-3V8Z\" fill=\"#fff\"/>\n<path d=\"M6 8h1v1H6V8Zm7 0h1v1h-1V8Z\" fill=\"#000\"/>\n</g>\n",
  'variant05': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M5 7h3v2H5V7Zm7 0h3v2h-3V7Z\" fill=\"#fff\"/>\n<path d=\"M5 8h2v1H5V8Zm7 0h2v1h-2V8Z\" fill=\"#000\"/>\n</g>\n",
  'variant04': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M5 7h3v3H5V7Zm7 0h3v3h-3V7Z\" fill=\"#fff\"/>\n<path d=\"M6 8h1v1H6V8Zm7 0h1v1h-1V8Z\" fill=\"#000\"/>\n</g>\n",
  'variant03': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M5 7h3v3H5V7Zm7 0h3v3h-3V7Z\" fill=\"#fff\"/>\n<path d=\"M6 7h2v2H6V7Zm7 0h2v2h-2V7Z\" fill=\"#000\"/>\n<path d=\"M6 7v1h1v1h1V8H7V7H6Zm7 0v1h1v1h1V8h-1V7h-1Z\" fill=\"#fff\" fill-opacity=\".4\"/>\n<path d=\"M7 7v1h1V7H7Zm7 0h1v1h-1V7Z\" fill=\"#fff\" fill-opacity=\".7\"/>\n</g>\n",
  'variant02': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M5 7h3v3H5V7Zm7 0h3v3h-3V7Z\" fill=\"#fff\"/>\n<path d=\"M5 8h2v1H5V8Zm7 0h2v1h-2V8Z\" fill=\"#000\"/>\n<path d=\"M5 8h1v1H5V8Zm7 0h1v1h-1V8Z\" fill=\"#fff\" fill-opacity=\".7\"/>\n</g>\n",
  'variant01': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M6 7h1v2H5V8h1V7Zm7 0h1v2h-2V8h1V7Z\" fill=\"#fff\"/>\n<path d=\"M7 7v1H6v1h2V7H7Zm7 0v1h-1v1h2V7h-1Z\" fill=\"#000\"/>\n<path d=\"M7 7v1h1V7H7ZM6 8v1h1V8H6Zm8-1v1h1V7h-1Zm-1 1v1h1V8h-1Z\" fill=\"#fff\" fill-opacity=\".5\"/>\n</g>\n"
};

const eyebrows = {
  'variant13': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M8 4v1H7v1H5V5h2V4h1Zm4 0h1v1h2v1h-2V5h-1V4Z\" fill=\"".concat(colors.hair.value, "\"/>\n<path d=\"M8 4v1H7v1H5V5h2V4h1Zm4 0h1v1h2v1h-2V5h-1V4Z\" fill=\"#000\" fill-opacity=\".1\"/>\n</g>\n"),
  'variant12': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M6 5h3v2H8V6H6V5Zm5 0h3v1h-2v1h-1V5Z\" fill=\"".concat(colors.hair.value, "\"/>\n<path d=\"M6 5h3v2H8V6H6V5Zm5 0h3v1h-2v1h-1V5Z\" fill=\"#000\" fill-opacity=\".1\"/>\n</g>\n"),
  'variant11': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M5 5h3v1h1v1H8V6H5V5Zm10 0h-3v1h-1v1h1V6h3V5Z\" fill=\"".concat(colors.hair.value, "\"/>\n<path d=\"M5 5h3v1h1v1H8V6H5V5Zm10 0h-3v1h-1v1h1V6h3V5Z\" fill=\"#000\" fill-opacity=\".1\"/>\n</g>\n"),
  'variant10': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M6 5H4v2h1V6h1V5Zm8 0h2v2h-1V6h-1V5Z\" fill=\"".concat(colors.hair.value, "\"/>\n<path d=\"M6 5H4v2h1V6h1V5Zm8 0h2v2h-1V6h-1V5Z\" fill=\"#000\" fill-opacity=\".1\"/>\n</g>\n"),
  'variant09': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M5 6h2v1H5V6Zm8 0h2v1h-2V6Z\" fill=\"".concat(colors.hair.value, "\"/>\n<path d=\"M5 6h2v1H5V6Zm8 0h2v1h-2V6Z\" fill=\"#000\" fill-opacity=\".1\"/>\n</g>\n"),
  'variant08': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M6 5h1v1h1v1H5V6h1V5Zm7 0h1v1h1v1h-3V6h1V5Z\" fill=\"".concat(colors.hair.value, "\"/>\n<path d=\"M6 5h1v1h1v1H5V6h1V5Zm7 0h1v1h1v1h-3V6h1V5Z\" fill=\"#000\" fill-opacity=\".1\"/>\n</g>\n"),
  'variant07': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M6 5h2v1h1v1H8V6H6V5Zm8 0h-2v1h-1v1h1V6h2V5Z\" fill=\"".concat(colors.hair.value, "\"/>\n<path d=\"M6 5h2v1h1v1H8V6H6V5Zm8 0h-2v1h-1v1h1V6h2V5Z\" fill=\"#000\" fill-opacity=\".1\"/>\n</g>\n"),
  'variant06': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M12 7V6h1V5h1v1h1v1h-1V6h-1v1h-1ZM5 7V6h1V5h1v1h1v1H7V6H6v1H5Z\" fill=\"".concat(colors.hair.value, "\"/>\n<path d=\"M12 7V6h1V5h1v1h1v1h-1V6h-1v1h-1ZM5 7V6h1V5h1v1h1v1H7V6H6v1H5Z\" fill=\"#000\" fill-opacity=\".1\"/>\n</g>\n"),
  'variant05': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M7 5v1H5v1H4V6h1V5h2Zm6 0h2v1h1v1h-1V6h-2V5Z\" fill=\"".concat(colors.hair.value, "\"/>\n<path d=\"M7 5v1H5v1H4V6h1V5h2Zm6 0h2v1h1v1h-1V6h-2V5Z\" fill=\"#000\" fill-opacity=\".1\"/>\n</g>\n"),
  'variant04': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M4 7V5h3v1H5v1H4Zm12-2v2h-1V6h-2V5h3Z\" fill=\"".concat(colors.hair.value, "\"/>\n<path d=\"M4 7V5h3v1H5v1H4Zm12-2v2h-1V6h-2V5h3Z\" fill=\"#000\" fill-opacity=\".1\"/>\n</g>\n"),
  'variant03': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M7 5h1v1h1v1H8V6H7V5Zm6 0v1h-1v1h-1V6h1V5h1Z\" fill=\"".concat(colors.hair.value, "\"/>\n<path d=\"M7 5h1v1h1v1H8V6H7V5Zm6 0v1h-1v1h-1V6h1V5h1Z\" fill=\"#000\" fill-opacity=\".1\"/>\n</g>\n"),
  'variant02': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M4 7V6h1V5h1v1H5v1H4Zm10-2h1v1h1v1h-1V6h-1V5Z\" fill=\"".concat(colors.hair.value, "\"/>\n<path d=\"M4 7V6h1V5h1v1H5v1H4Zm10-2h1v1h1v1h-1V6h-1V5Z\" fill=\"#000\" fill-opacity=\".1\"/>\n</g>\n"),
  'variant01': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M7 5v1H5v1H4V6h1V5h2Zm7 0v1h-2v1h-1V6h1V5h2Z\" fill=\"".concat(colors.hair.value, "\"/>\n<path d=\"M7 5v1H5v1H4V6h1V5h2Zm7 0v1h-2v1h-1V6h1V5h2Z\" fill=\"#000\" fill-opacity=\".1\"/>\n</g>\n")
};

const mouth$1 = {
  'surprised03': (components, colors) => "\n<path d=\"M9 12v2h2v-2H9Z\" fill=\"".concat(colors.mouth.value, "\"/>\n"),
  'surprised02': (components, colors) => "\n<path d=\"M9 13v1h1v-1H9Z\" fill=\"".concat(colors.mouth.value, "\"/>\n"),
  'happy09': (components, colors) => "\n<path d=\"M7 12v1h1v1h4v-1H8v-1H7Z\" fill=\"".concat(colors.mouth.value, "\"/>\n"),
  'happy08': (components, colors) => "\n<path d=\"M10 12v1H9v1h2v-2h-1Z\" fill=\"".concat(colors.mouth.value, "\"/>\n"),
  'happy07': (components, colors) => "\n<path d=\"M8 13v1h4v-1h1v-1h-1v1H8Z\" fill=\"".concat(colors.mouth.value, "\"/>\n"),
  'happy06': (components, colors) => "\n<path d=\"M9 12v2h2v-1h-1v-1H9Z\" fill=\"".concat(colors.mouth.value, "\"/>\n<path d=\"M11 14v-1h-1v-1H9v1h1v1h1Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n"),
  'happy05': (components, colors) => "\n<path d=\"M8 12v1h1v1h2v-1h1v-1H8Z\" fill=\"".concat(colors.mouth.value, "\"/>\n"),
  'happy04': (components, colors) => "\n<path d=\"M9 13v1h2v-1h1v-1h-1v1H9Z\" fill=\"".concat(colors.mouth.value, "\"/>\n"),
  'happy03': (components, colors) => "\n<path d=\"M8 12v1h1v1h2v-1H9v-1H8Z\" fill=\"".concat(colors.mouth.value, "\"/>\n"),
  'happy02': (components, colors) => "\n<path d=\"M8 12v1h1v1h2v-1h1v-1h-1v-1H9v1H8Z\" fill=\"".concat(colors.mouth.value, "\"/>\n<path d=\"M9 12v1h2v-1H9Z\" fill=\"#fff\"/>\n"),
  'happy01': (components, colors) => "\n<path d=\"M8 12v1h1v1h2v-1h1v-1h-1v-1H9v1H8Z\" fill=\"".concat(colors.mouth.value, "\"/>\n<path d=\"M9 12v1h2v-1H9Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n"),
  'sad08': (components, colors) => "\n<path d=\"M9 12v1H8v1h4v-1h-1v-1H9Z\" fill=\"".concat(colors.mouth.value, "\"/>\n"),
  'sad07': (components, colors) => "\n<path d=\"M11 12v1H9v1H8v-1h1v-1h2Z\" fill=\"".concat(colors.mouth.value, "\"/>\n"),
  'sad06': (components, colors) => "\n<path d=\"M8 12h3v1H8v-1Z\" fill=\"".concat(colors.mouth.value, "\"/>\n"),
  'sad05': (components, colors) => "\n<path d=\"M8 12h4v1H8v-1Z\" fill=\"".concat(colors.mouth.value, "\"/>\n"),
  'sad04': (components, colors) => "\n<path d=\"M9 12h2v1H9v-1Z\" fill=\"".concat(colors.mouth.value, "\"/>\n"),
  'sad03': (components, colors) => "\n<path d=\"M8 12v1h3v1h1v-1h-1v-1H8Z\" fill=\"".concat(colors.mouth.value, "\"/>\n"),
  'sad02': (components, colors) => "\n<path d=\"M8 13v1h1v-1h3v-1H9v1H8Z\" fill=\"".concat(colors.mouth.value, "\"/>\n"),
  'sad01': (components, colors) => "\n<path d=\"M10 12v1h1v1h1v-2h-2Z\" fill=\"".concat(colors.mouth.value, "\"/>\n"),
  'surprised01': (components, colors) => "\n<path d=\"M8 12v2h4v-2H8Z\" fill=\"#fff\"/>\n"
};

const hair$1 = {
  'short11': (components, colors) => "\n<path d=\"M4 4v2h1V5h1V4h8v1h1v1h1V4h-1V3H5v1H4Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'short10': (components, colors) => "\n<path d=\"M6 2h8v1h1v1h1v3h-1V6h-1V5h-1V4H9v1H7v1H6v1H4V4h1V3h1V2Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'short09': (components, colors) => "\n<path d=\"M4 7h1V5h1V4h2v1h1v1h2V5h1V4h2v1h1v2h1V5h-1V4h-1V3H6v1H5v1H4v2Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'short08': (components, colors) => "\n<path d=\"M3 8h1V5h12v3h1V4h-1V3h-1V2H5v1H4v1H3v4Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'short07': (components, colors) => "\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M2 5v1h1v1h2V5h1V3H4v1H3v1H2Zm6-1h2v1h2V4h1V3H8v1Zm6 1h1v2h2V6h1V5h-1V4h-1V3h-2v2Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'short06': (components, colors) => "\n<path d=\"M3 8h1V5h2V3h8v1h1v1h1v3h1V3h-2V2h-2V1h-1v1h-2V1H9v1H8V1H7v1H5v1H4v1H3v4Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'short05': (components, colors) => "\n<path d=\"M11 5V4h1V3H8v1h1v1h2Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'short04': (components, colors) => "\n<path d=\"M3 8h1V6h2V5h2V4h1v1h2v1h2v1h2v1h2V5h-1V4h-1V3H5v1H4v1H3v3Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'short03': (components, colors) => "\n<path d=\"M5 5h10V4h-1V3H6v1H5v1Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'short02': (components, colors) => "\n<path d=\"M3 5H2v3h2V5h1V4h1v1h2V4h4V3h1v1h1v1h1v1h1v2h1V7h1V5h-1V4h-1V3h-1V2h-1V1h-1v1h-2V1h-1v1H7V1H5v1H4v1H3v2Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'long15': (components, colors) => "\n<path d=\"M1 7h1v5h1v1h1V7h1V6h1V4h1V3h6v1h1v1h1v1h1v7h1v-2h1V7h-1V6h1V4h-1v1h-1V3h1V2h-1v1h-1V2h-1V1h-1V0h-1v1H6V0H5v1H4V0H3v1h1v2H3V2H2v1h1v1H2v2H1v1Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'short01': (components, colors) => "\n<path d=\"M6 3v1H5v1H4v3h2V7h1V6h4V5h1V4h1v1h-1v1h1v1h1v1h2V5h-1V4h-1V3H6Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'long14': (components, colors) => "\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M3 11v6h1v-1h4v-1H6v-1H5v-1H4v-2H3Zm0-3h2V5h10v3h2V4h-1V3H4v1H3v4Zm13 3h1v6h-1v-1h-4v-1h2v-1h1v-1h1v-2Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'long13': (components, colors) => "\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M3 11v5h2v-3H4v-2H3Zm0-3h2V5h10v3h2V4h-1V3H4v1H3v4Zm13 3h1v5h-2v-3h1v-2Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'long12': (components, colors) => "\n<path d=\"M4 13h1v1H3V4h1V3h1V2h10v1h1v1h1v10h-2v-1h1V5H4v8Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'long11': (components, colors) => "\n<path d=\"M2 18h2v-1h4v-2H6v-1H5v-1H4V5h1V4h1v1h1V4h1v1h1V4h6v1h1v8h-1v1h-1v1h-2v2h4v1h2V4h-1V3h-1V2H4v1H3v1H2v14Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'long10': (components, colors) => "\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M3 14.104V11h1v2h1v1l-2 .104ZM17 11v3h-2v-1h1v-2h1ZM3 8h1V5h1V4h1V3h8v1h1v1h1v3h1V4h-1V3h-1V2H5v1H4v1H3v4Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'long09': (components, colors) => "\n<path d=\"M2 15h2V5h12v10h2V4h-1V3h-1V2H4v1H3v1H2v11Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'long08': (components, colors) => "\n<path d=\"M2 14h2V7h1V6h1V5h1V4h1V3h6v1h1v1h1v9h2V3h-1V2h-1V1H4v1H3v1H2v11Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'long07': (components, colors) => "\n<path d=\"M4 5v3h1V5h2v1h1V5h1v1h1V5h5v3h1V5h1V3.016h-1V2h-2v1l-8 .016V2H4v1.016H3V5h1Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'long06': (components, colors) => "\n<path d=\"M3 11h1V9h1V8h1V7h1V6h1V5h5v1h3V5h-1V4h-1V3H5v1H4v1H3v6Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'long05': (components, colors) => "\n<path d=\"M2 17h2v-1h4v-1H6v-1H5v-1H3.998L4 11H3V8h1V5h1V4h6V3h1v1h1V3h1v1h1v1h1v3h1v3h-1v2h-1v1h-1v1h-2v1h4v1h2V4h-1V3h-1V2h-1V1H5v1H4v1H3v1H2v13Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'long04': (components, colors) => "\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M3 14h2v-1H4v-2H3v3Zm12 0h2v-3h-1v2h-1v1Zm2-6V3h-1V2H4v1H3v5h1V5h1V4h1V3h8v1h1v1h1v3h1Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'long03': (components, colors) => "\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M3 13v1H2v2H1v1h1v-1h1v-1h1v-1h1v-1H4v-2H3v2Zm0-5h1V7h1V6h1V5h8v1h1v1h1v1h1V4h-1V3h-1V2H5v1H4v1H3v4Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'long02': (components, colors) => "\n<path d=\"M2 15h4v-1H5v-1H4V6h2V5h2V4h1V3h4v1h1v1h1v1h1v7h-1v1h-1v1h4V4h-1V3h-1V2h-1V1H5v1H4v1H3v1H2v11Z\" fill=\"".concat(colors.hair.value, "\"/>\n"),
  'long01': (components, colors) => "\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M3 11v3H2v2H1v1h1v-1h1v-1h1v-1h1v-1H4v-2H3Zm14 0v3h1v2h1v1h-1v-1h-1v-1h-1v-1h-1v-1h1v-2h1Zm-1-3h1V4h-1V3h-1V2H5v1H4v1H3v4h1V7h1V6h1V5h8v1h1v1h1v1Z\" fill=\"".concat(colors.hair.value, "\"/>\n")
};

const accessories$1 = {
  'variant04': (components, colors) => "\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M3 10v1h1v-1H3Zm13 0v1h1v-1h-1Z\" fill=\"".concat(colors.accessories.value, "\"/>\n"),
  'variant03': (components, colors) => "\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M3 10v2h1v-2H3Zm13 0h1v2h-1v-2Z\" fill=\"".concat(colors.accessories.value, "\"/>\n"),
  'variant02': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M3 10v2h1v-2H3Zm13 0h1v2h-1v-2Z\" fill=\"".concat(colors.accessories.value, "\"/>\n<path d=\"M3 10v1h1v-1H3Zm13 0h1v1h-1v-1Z\" fill=\"#fff\" fill-opacity=\".4\"/>\n</g>\n"),
  'variant01': (components, colors) => "\n<g fill=\"".concat(colors.accessories.value, "\">\n<path d=\"M4 11H2v2h2v-2ZM18 11h-2v2h2v-2Z\"/>\n</g>\n")
};

const glasses$1 = {
  'variant07': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path opacity=\".2\" d=\"M5 7h3v3H5V7Zm7 0h3v3h-3V7Z\" fill=\"#fff\"/>\n<path opacity=\".2\" d=\"M7 7h1v1H7V7Zm7 0h1v1h-1V7Z\" fill=\"#fff\"/>\n<path d=\"M12 10V7h3v3h-3Zm-1-4v1H9l-.036-1H4v1H3v1h1v3h5V8h2v3h5V8h1V7h-1V6h-5Zm-6 4V7h3v3H5Z\" fill=\"".concat(colors.glasses.value, "\"/>\n<path d=\"M3 7h1v1H3V7Zm6 0h2v1H9V7Zm7 0h1v1h-1V7Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n</g>\n"),
  'variant06': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path opacity=\".2\" d=\"M5 7h3v2H5V7Zm7 0h3v2h-3V7Z\" fill=\"#fff\"/>\n<path opacity=\".2\" d=\"M7 7h1v1H7V7Zm7 0h1v1h-1V7Z\" fill=\"#fff\"/>\n<path d=\"M5 7v2h3V7H5ZM4 6v1H3v1h1v1h1v1h3V9h1V8h2v1h1v1h3V9h1V8h1V7h-1V6h-5v1H9V6H4Zm8 1v2h3V7h-3Z\" fill=\"".concat(colors.glasses.value, "\"/>\n<path d=\"M3 7h1v1H3V7Zm6 0h2v1H9V7Zm7 0h1v1h-1V7Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n</g>\n"),
  'variant05': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path opacity=\".2\" d=\"M5 8h3v1H5V8Zm7 0h3v1h-3V8Z\" fill=\"#fff\"/>\n<path opacity=\".2\" d=\"M7 8h1v1H7V8Zm7 0h1v1h-1V8Z\" fill=\"#fff\"/>\n<path d=\"M5 8v1h3V8H5ZM3 7v1h1v1h1v1h3V9h1V8h2v1h1v1h3V9h1V8h1V7H3Zm9 1v1h3V8h-3Z\" fill=\"".concat(colors.glasses.value, "\"/>\n<path d=\"M3 7v1h1V7H3Zm6 0v1h2V7H9Zm7 0v1h1V7h-1Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n</g>\n"),
  'variant04': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path opacity=\".2\" d=\"M5 7h3v2H5V7Zm7 0h3v2h-3V7Z\" fill=\"#fff\"/>\n<path opacity=\".2\" d=\"M7 7h1v1H7V7Zm7 0h1v1h-1V7Z\" fill=\"#fff\"/>\n<path d=\"M12 7v2h3V7h-3ZM8 6H5v1H3v1h1v1h1v1h3V9h1V8h2v1h1v1h3V9h1V8h1V7h-2V6h-3v1H8V6ZM5 7v2h3V7H5Z\" fill=\"".concat(colors.glasses.value, "\"/>\n<path d=\"M3 7h1v1H3V7Zm6 0h2v1H9V7Zm7 0h1v1h-1V7Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n</g>\n"),
  'variant03': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M4 8H3V7h1V6h5v1h2V6h5v1h1v1h-1v2h-5V8H9v2H4V8Zm1 0V7h3v2H5V8Zm7-1v2h3V7h-3Z\" fill=\"".concat(colors.glasses.value, "\"/>\n<path opacity=\".2\" d=\"M5 7h3v2H5V7Zm7 0h3v2h-3V7Z\" fill=\"#fff\"/>\n<path opacity=\".2\" d=\"M14 7h1v1h-1V7ZM7 7h1v1H7V7Z\" fill=\"#fff\"/>\n<path d=\"M3 8V7h1v1H3Zm6-1v1h2V7H9Zm7 0v1h1V7h-1Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n</g>\n"),
  'variant02': (components, colors) => "\n<g fill-rule=\"evenodd\" clip-rule=\"evenodd\">\n<path d=\"M4 8H3V7h14v1h-1v2h-5V8H9v2H4V8Zm1 0h3v1H5V8Zm7 0h3v1h-3V8Z\" fill=\"".concat(colors.glasses.value, "\"/>\n<path opacity=\".2\" d=\"M7 8v1h1V8H7Zm7 0v1h1V8h-1Z\" fill=\"#fff\"/>\n<path opacity=\".2\" d=\"M5 8h3v1H5V8Zm7 0h3v1h-3V8Z\" fill=\"#fff\"/>\n<path d=\"M3 7v1h1V7H3Zm13 0v1h1V7h-1ZM9 7v1h2V7H9Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n</g>\n"),
  'variant01': (components, colors) => "\n<path d=\"M3 8V7h1V6h2v1h1V6h2v1h2V6h2v1h1V6h2v1h1v1h-1v1h-1v1h-1v1h-1v-1h-1V9h-1V8H9v1H8v1H7v1H6v-1H5V9H4V8H3Z\" fill=\"".concat(colors.glasses.value, "\"/>\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M3 7v1h1V7h1V6H4v1H3Zm5-1v1h1v1h2V7h1V6h-1v1H9V6H8Zm7 0v1h1v1h1V7h-1V6h-1Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n")
};

const hat$1 = {
  'variant01': (components, colors) => "\n<path d=\"M4 1v2H2v2h16V3h-2V1H4Z\" fill=\"".concat(colors.hat.value, "\"/>\n"),
  'variant02': (components, colors) => "\n<path d=\"M4 4H2v1h16V4h-2V1H4v3Z\" fill=\"".concat(colors.hat.value, "\"/>\n"),
  'variant03': (components, colors) => "\n<path d=\"M3 4v2h14V4h-1V3h-1V2H5v1H4v1H3Z\" fill=\"".concat(colors.hat.value, "\"/>\n"),
  'variant04': (components, colors) => "\n<path d=\"M6 1v1H5v1H4v2h14V4h-2V3h-1V2h-1V1H6Z\" fill=\"".concat(colors.hat.value, "\"/>\n"),
  'variant05': (components, colors) => "\n<path d=\"M3 5V3h2V1h10v2h2v2H3Z\" fill=\"".concat(colors.hat.value, "\"/>\n<path d=\"M4 1v2h12V1H4Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n"),
  'variant06': (components, colors) => "\n<path d=\"M3 5V4h2V1h10v3h2v1H3Z\" fill=\"".concat(colors.hat.value, "\"/>\n<path d=\"M5 1v3h10V1H5Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n"),
  'variant07': (components, colors) => "\n<path d=\"M3 6V4h1V3h1V2h10v1h1v1h1v2H3Z\" fill=\"".concat(colors.hat.value, "\"/>\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M4 3v1h12V3H4ZM3 5v1h14V5H3Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n"),
  'variant08': (components, colors) => "\n<path d=\"M14 2H6v1H5v1H4v2h14V5h-2V4h-1V3h-1V2Z\" fill=\"".concat(colors.hat.value, "\"/>\n<path d=\"M5 6h1V5h1V4h1V3H7v1H6v1H5v1Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n"),
  'variant09': (components, colors) => "\n<path d=\"M5 1v2H3v2h14V3h-2V1H5Z\" fill=\"".concat(colors.hat.value, "\"/>\n<path d=\"M14 3V0h-1v3h1Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n"),
  'variant10': (components, colors) => "\n<path d=\"M4 0v5H2v1h16V5h-2V0H4Z\" fill=\"".concat(colors.hat.value, "\"/>\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M15 3V0h-1v3h1Zm-2-3v2h-1V0h1Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n"),
  'variant11': (components, colors) => "\n<path d=\"M3 3v2h14V3h-1V2h-1V1H5v1H4v1H3Z\" fill=\"".concat(colors.hat.value, "\"/>\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M14 1v4h-1V1h1Zm-2 0v4h-1V1h1Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n"),
  'variant12': (components, colors) => "\n<path d=\"M5 4H4v2h14V5h-2V4h-1V3h-1V2H6v1H5v1Z\" fill=\"".concat(colors.hat.value, "\"/>\n<path d=\"M14 4h-3v1h3V4Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n")
};

const clothing = {
  'variant25': (components, colors) => "\n<path d=\"M3 20v-3h1v-1h12v1h1v3H3Z\" fill=\"".concat(colors.clothes.value, "\"/>\n<path d=\"M5 20v-2h1v-1h8v1h1v2h-2v-1h-2v1H9v-1H7v1H5Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n"),
  'variant24': (components, colors) => "\n<path d=\"M3 20v-3h1v-1h12v1h1v3H3Z\" fill=\"".concat(colors.clothes.value, "\"/>\n<path d=\"M8 16H4v1H3v3h14v-3h-1v-1h-4v1h1v1h-1v1h-1v-1H9v1H8v-1H7v-1h1v-1Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n<path d=\"M9 16v1h2v-1H9Z\" fill=\"#fff\"/>\n"),
  'variant23': (components, colors) => "\n<path d=\"M3 20v-3h1v-1h12v1h1v3H3Z\" fill=\"".concat(colors.clothes.value, "\"/>\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M9 16H4v1H3v3h6v-2H8v-1h1v-1Zm2 0h5v1h1v3h-6v-2h1v-1h-1v-1Z\" fill=\"#fff\" fill-opacity=\".3\"/>\n"),
  'variant22': (components, colors) => "\n<path d=\"M3 20v-3h1v-1h3v2h6v-2h3v1h1v3H3Z\" fill=\"".concat(colors.clothes.value, "\"/>\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M5 16H4v1H3v3h2v-4Zm1 0h1v2h6v-2h1v4H6v-4Zm9 0h1v1h1v3h-2v-4Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n"),
  'variant21': (components, colors) => "\n<path d=\"M3 20v-3h1v-1h4v1h1v1h2v-1h1v-1h4v1h1v3H3Z\" fill=\"".concat(colors.clothes.value, "\"/>\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M4 17v-1h3v1H4Zm9 0v-1h3v1h-3Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n"),
  'variant20': (components, colors) => "\n<path d=\"M3 20v-3h1v-1h3v-1h1v1h1v1h2v-1h.965L12 15h1v1h3v1h1v3H3Z\" fill=\"".concat(colors.clothes.value, "\"/>\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M6 16H4v1H3v3h6v-2H8v-1H6v-1Zm2 0h1-1Zm3 0h1-1Zm2 0h1v1h-2v1h-1v2h6v-3h-1v-1h-3Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n"),
  'variant19': (components, colors) => "\n<path d=\"M5 16H4v1H3v3h14v-3h-1v-1h-3v1H7v-1H5Z\" fill=\"".concat(colors.clothes.value, "\"/>\n<path d=\"M10 20v-1h3v1h-3Z\" fill=\"#fff\" fill-opacity=\".5\"/>\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M5 16H4v1H3v3h1v-1h1v-3Zm1 0h1v1h6v-1h1v2H6v-2Zm9 0h1v1h1v3h-1v-1h-1v-3Z\" fill=\"#fff\" fill-opacity=\".8\"/>\n"),
  'variant18': (components, colors) => "\n<path d=\"M3 20v-3h1v-1h4v1h4v-1h4v1h1v3H3Z\" fill=\"".concat(colors.clothes.value, "\"/>\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M3 20v-1h1v1H3Zm2 0v-1h1v1H5Zm2 0v-1h1v1H7Zm2 0v-1h1v1H9Zm2 0v-1h1v1h-1Zm2 0v-1h1v1h-1Zm2 0v-1h1v1h-1Zm1-2h1v1h-1v-1Zm-2 0h1v1h-1v-1Zm-2 0h1v1h-1v-1Zm-2 0h1v1h-1v-1Zm-2 0h1v1H8v-1Zm-2 0h1v1H6v-1Zm-2 0h1v1H4v-1Zm-1-1h1v1H3v-1Zm2 0h1v1H5v-1Zm2 0h1v1H7v-1Zm2 0h1v1H9v-1Zm2 0h1v1h-1v-1Zm2 0h1v1h-1v-1Zm2 0h1v1h-1v-1ZM4 16h1v1H4v-1Zm2 0h1v1H6v-1Zm6 0h1v1h-1v-1Zm2 0h1v1h-1v-1Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n"),
  'variant17': (components, colors) => "\n<path d=\"M3 20v-3h1v-1h4v1h4v-1h4v1h1v3H3Z\" fill=\"".concat(colors.clothes.value, "\"/>\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M3 20v-2h1v2H3Zm3 0v-2h2v2H6Zm4 0v-2h2v2h-2Zm4 0v-2h2v2h-2Zm2-3v1h1v-1h-1Zm-2 1v-2h-2v2h2Zm-6-1v1h2v-1H8Zm-4-1v2h2v-2H4Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n"),
  'variant16': (components, colors) => "\n<path d=\"M3 20v-3h1v-1h4v1h4v-1h4v1h1v3H3Z\" fill=\"".concat(colors.clothes.value, "\"/>\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M3 19h14v1H3v-1Zm0-2h14v1H3v-1Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n"),
  'variant15': (components, colors) => "\n<path d=\"M3 20v-3h1v-1h4v1h4v-1h4v1h1v3H3Z\" fill=\"".concat(colors.clothes.value, "\"/>\n"),
  'variant14': (components, colors) => "\n<path d=\"M3 20v-3h1v-1h4.027L8 15h4l.011 1H16v1h1v3H3Z\" fill=\"".concat(colors.clothes.value, "\"/>\n<path d=\"M3 20v-3h1v-1h12v1h1v3H3Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n<path d=\"M12 20v-1h3v1h-3Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n"),
  'variant13': (components, colors) => "\n<path d=\"M3 20v-3h1v-1h12v1h1v3H3Z\" fill=\"".concat(colors.clothes.value, "\"/>\n"),
  'variant12': (components, colors) => "\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M4 16v4h4v-1H7v-1H6v-1H5v-1H4Zm12 0v4h-4v-1h1v-1h1v-1h1v-1h1Z\" fill=\"".concat(colors.clothes.value, "\"/>\n"),
  'variant11': (components, colors) => "\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M5 16h1v2h1v1h1v1H5v-4Zm9 0h1v4h-3v-1h1v-1h1v-2Z\" fill=\"".concat(colors.clothes.value, "\"/>\n"),
  'variant10': (components, colors) => "\n<path d=\"M4 20v-2h1v-1h1v-1h2v1h1v1h2v-1h1v-1h2v1h1v1h1v2H4Z\" fill=\"".concat(colors.clothes.value, "\"/>\n"),
  'variant09': (components, colors) => "\n<path d=\"M3 20v-3h1v-1h4v1h1v1h2v-1h1v-1h4v1h1v3H3Z\" fill=\"".concat(colors.clothes.value, "\"/>\n<path d=\"M3 20v-3h1v-1h2v1h1v1h1v1h4v-1h1v-1h1v-1h2v1h1v3H3Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n"),
  'variant08': (components, colors) => "\n<path d=\"M3 20v-3h1v-1h5v1h2v-1h5v1h1v3H3Z\" fill=\"".concat(colors.clothes.value, "\"/>\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M3 20v-2h1v1h3v1H3Zm14 0v-2h-1v1h-3v1h4Z\" fill=\"#fff\" fill-opacity=\".4\"/>\n<path d=\"M7 16H4v1H3v1h1v1h3v1h6v-1h3v-1h1v-1h-1v-1h-3v1H7v-1Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n"),
  'variant07': (components, colors) => "\n<path d=\"M3 20v-3h1v-1h4v1h4v-1h4v1h1v3H3Z\" fill=\"".concat(colors.clothes.value, "\"/>\n<path d=\"M15 20h2v-3h-1v-1h-4v1H8v-1H4v1H3v3h2v-2h10v2Z\" fill=\"#fff\" fill-opacity=\".4\"/>\n"),
  'variant06': (components, colors) => "\n<path d=\"M3 20v-3h1v-1h4v1h1v1h2v-1h1v-1h4v1h1v3H3Z\" fill=\"".concat(colors.clothes.value, "\"/>\n<path d=\"M6 16v1h1v1h1v1h4v-1h1v-1h1v-1h-2v1h-1v1H9v-1H8v-1H6Z\" fill=\"#fff\" fill-opacity=\".4\"/>\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M13 20v-1h2v1h-2Zm1-4v1h2v-1h-2Zm-8 0H4v1h2v-1Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n"),
  'variant05': (components, colors) => "\n<path d=\"M3 20v-3h1v-1h3v1h1v1h1v1h2v-1h1v-1h1v-1h3v1h1v3H3Z\" fill=\"".concat(colors.clothes.value, "\"/>\n"),
  'variant04': (components, colors) => "\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M3 20v-3h1v-1h2v1h1v1h1v1h1v1H3Zm14 0v-3h-1v-1h-2v1h-1v1h-1v1h-1v1h6Z\" fill=\"".concat(colors.clothes.value, "\"/>\n"),
  'variant03': (components, colors) => "\n<g fill=\"".concat(colors.clothes.value, "\">\n<path d=\"M4 16v4h4v-1H7v-1H6v-2H4ZM16 20v-4h-2v2h-1v1h-1v1h4Z\"/>\n</g>\n"),
  'variant02': (components, colors) => "\n<path d=\"M3 20v-3h1v-1h3v1h1v1h1v1h2v-1h1v-1h1v-1h3v1h1v3H3Z\" fill=\"".concat(colors.clothes.value, "\"/>\n<path d=\"M6 16v1h1v1h1v1h1v1h2v-1h1v-1h1v-1h1v-1h-1v1h-1v1h-1v1H9v-1H8v-1H7v-1H6Z\" fill=\"#fff\" fill-opacity=\".4\"/>\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M15 16v1h-1v1h-1v1h-1v1h-1v-1h1v-1h1v-1h1v-1h1ZM5 16v1h1v1h1v1h1v1h1v-1H8v-1H7v-1H6v-1H5Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M3 20h1v-3h1v1h1v1h1v1h1v-1H7v-1H6v-1H5v-1H4v1H3v3Zm14 0v-3h-1v-1h-1v1h-1v1h-1v1h-1v1h1v-1h1v-1h1v-1h1v3h1Z\" fill=\"#fff\" fill-opacity=\".1\"/>\n"),
  'variant01': (components, colors) => "\n<path d=\"M3 20v-3h1v-1h5v1h2v-1h5v1h1v3H3Z\" fill=\"".concat(colors.clothes.value, "\"/>\n<path fill-rule=\"evenodd\" clip-rule=\"evenodd\" d=\"M3 17h14v1H3v-1Zm0 2v1h14v-1H3Z\" fill=\"#fff\" fill-opacity=\".2\"/>\n")
};

var components = /*#__PURE__*/Object.freeze({
  __proto__: null,
  beard: beard,
  eyes: eyes,
  eyebrows: eyebrows,
  mouth: mouth$1,
  hair: hair$1,
  accessories: accessories$1,
  glasses: glasses$1,
  hat: hat$1,
  clothing: clothing
});

function pickComponent(prng, group) {
  let values = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : [];
  const componentCollection = components;
  const key = prng.pick(values);

  if (componentCollection[group][key]) {
    return {
      name: key,
      value: componentCollection[group][key]
    };
  } else {
    return undefined;
  }
}

const skin = {
  'variant01': 'rgba(255, 219, 172, 1)',
  'variant02': 'rgba(245, 207, 160, 1)',
  'variant03': 'rgba(234, 195, 147, 1)',
  'variant04': 'rgba(224, 182, 135, 1)',
  'variant05': 'rgba(203, 158, 110, 1)',
  'variant06': 'rgba(182, 134, 85, 1)',
  'variant07': 'rgba(162, 109, 61, 1)',
  'variant08': 'rgba(141, 85, 36, 1)'
};

const hair = {
  'variant01': 'rgba(202, 177, 136, 1)',
  'variant02': 'rgba(167, 137, 97, 1)',
  'variant03': 'rgba(131, 98, 59, 1)',
  'variant04': 'rgba(96, 58, 20, 1)',
  'variant05': 'rgba(96, 48, 21, 1)',
  'variant06': 'rgba(97, 38, 22, 1)',
  'variant07': 'rgba(97, 28, 23, 1)',
  'variant08': 'rgba(78, 26, 19, 1)',
  'variant09': 'rgba(59, 23, 14, 1)',
  'variant10': 'rgba(40, 21, 10, 1)'
};

const accessories = {
  'gold01': 'rgba(218, 165, 32, 1)',
  'gold02': 'rgba(255, 215, 0, 1)',
  'gold03': 'rgba(250, 250, 210, 1)',
  'silver01': 'rgba(211, 211, 211, 1)',
  'silver02': 'rgba(169, 169, 169, 1)'
};

const mouth = {
  'variant01': 'rgba(210, 153, 133, 1)',
  'variant02': 'rgba(201, 130, 118, 1)',
  'variant03': 'rgba(227, 93, 106, 1)',
  'variant04': 'rgba(222, 15, 13, 1)'
};

const clothes = {
  'yellow01': 'rgba(255, 238, 173, 1)',
  'yellow02': 'rgba(255, 217, 105, 1)',
  'yellow03': 'rgba(255, 196, 37, 1)',
  'red01': 'rgba(255, 111, 105, 1)',
  'red02': 'rgba(209, 17, 65, 1)',
  'red03': 'rgba(174, 0, 1, 1)',
  'blue01': 'rgba(91, 192, 222, 1)',
  'blue02': 'rgba(66, 139, 202, 1)',
  'blue03': 'rgba(3, 57, 108, 1)',
  'green01': 'rgba(136, 216, 176, 1)',
  'green02': 'rgba(68, 197, 133, 1)',
  'green03': 'rgba(0, 177, 89, 1)'
};

const hat = {
  'pink01': 'rgba(204, 97, 146, 1)',
  'blue01': 'rgba(38, 99, 163, 1)',
  'green01': 'rgba(61, 138, 107, 1)',
  'purple01': 'rgba(97, 79, 138, 1)',
  'red01': 'rgba(166, 33, 22, 1)',
  'black01': 'rgba(46, 30, 5, 1)',
  'gray01': 'rgba(152, 151, 137, 1)'
};

const glasses = {
  'green01': 'rgba(95, 112, 92, 1)',
  'blue01': 'rgba(67, 103, 125, 1)',
  'red01': 'rgba(160, 75, 93, 1)',
  'black01': 'rgba(75, 75, 75, 1)',
  'black02': 'rgba(50, 50, 50, 1)',
  'black03': 'rgba(25, 25, 25, 1)'
};

var colors = /*#__PURE__*/Object.freeze({
  __proto__: null,
  skin: skin,
  hair: hair,
  accessories: accessories,
  mouth: mouth,
  clothes: clothes,
  hat: hat,
  glasses: glasses
});

function pickColor(prng, group, values) {
  var _colorCollection$grou;

  const colorCollection = colors;

  if (values.length === 0) {
    values.push('transparent');
  }

  const key = prng.pick(values);
  return {
    name: key,
    value: (_colorCollection$grou = colorCollection[group][key]) !== null && _colorCollection$grou !== void 0 ? _colorCollection$grou : key
  };
}

const style = {
  meta: {
    title: 'Pixel Art',
    creator: 'Plastic Jam',
    license: {
      name: 'MIT',
      url: 'https://github.com/dicebear/dicebear/blob/main/packages/%40dicebear/pixel-art/LICENSE'
    }
  },
  schema: schema$1,
  create: _ref => {
    var _options$skinColor, _options$hairColor, _options$accessoriesC, _options$mouthColor, _options$clothesColor, _options$hatColor, _options$glassesColor, _components$beard$val, _components$beard, _components$eyes$valu, _components$eyes, _components$eyebrows$, _components$eyebrows, _components$mouth$val, _components$mouth, _components$hair$valu, _components$hair, _components$accessori, _components$accessori2, _components$glasses$v, _components$glasses, _components$hat$value, _components$hat, _components$clothing$, _components$clothing;

    let {
      prng,
      options
    } = _ref;
    const beardComponent = pickComponent(prng, 'beard', options.beard);
    const eyesComponent = pickComponent(prng, 'eyes', options.eyes);
    const eyebrowsComponent = pickComponent(prng, 'eyebrows', options.eyebrows);
    const mouthComponent = pickComponent(prng, 'mouth', options.mouth);
    const hairComponent = pickComponent(prng, 'hair', options.hair);
    const accessoriesComponent = pickComponent(prng, 'accessories', options.accessories);
    const glassesComponent = pickComponent(prng, 'glasses', options.glasses);
    const hatComponent = pickComponent(prng, 'hat', options.hat);
    const clothingComponent = pickComponent(prng, 'clothing', options.clothing);
    const components = {
      'beard': prng.bool(options.beardProbability) ? beardComponent : undefined,
      'eyes': eyesComponent,
      'eyebrows': eyebrowsComponent,
      'mouth': mouthComponent,
      'hair': prng.bool(options.hairProbability) ? hairComponent : undefined,
      'accessories': prng.bool(options.accessoriesProbability) ? accessoriesComponent : undefined,
      'glasses': prng.bool(options.glassesProbability) ? glassesComponent : undefined,
      'hat': prng.bool(options.hatProbability) ? hatComponent : undefined,
      'clothing': clothingComponent
    };
    const colors = {
      'skin': pickColor(prng, 'skin', (_options$skinColor = options.skinColor) !== null && _options$skinColor !== void 0 ? _options$skinColor : []),
      'hair': pickColor(prng, 'hair', (_options$hairColor = options.hairColor) !== null && _options$hairColor !== void 0 ? _options$hairColor : []),
      'accessories': pickColor(prng, 'accessories', (_options$accessoriesC = options.accessoriesColor) !== null && _options$accessoriesC !== void 0 ? _options$accessoriesC : []),
      'mouth': pickColor(prng, 'mouth', (_options$mouthColor = options.mouthColor) !== null && _options$mouthColor !== void 0 ? _options$mouthColor : []),
      'clothes': pickColor(prng, 'clothes', (_options$clothesColor = options.clothesColor) !== null && _options$clothesColor !== void 0 ? _options$clothesColor : []),
      'hat': pickColor(prng, 'hat', (_options$hatColor = options.hatColor) !== null && _options$hatColor !== void 0 ? _options$hatColor : []),
      'glasses': pickColor(prng, 'glasses', (_options$glassesColor = options.glassesColor) !== null && _options$glassesColor !== void 0 ? _options$glassesColor : [])
    };
    return {
      attributes: {
        viewBox: '0 0 20 20',
        fill: 'none'
      },
      body: "\n<path d=\"M6 4V3h8v1h1v1h1v3h1v3h-1v2h-1v1h-1v1h-2v1h4v1h1v3H3v-3h1v-1h4v-1H6v-1H5v-1H4v-2H3V8h1V5h1V4h1Z\" fill=\"".concat(colors.skin.value, "\"/>\n<path d=\"M6 3v1H5v1H4v3H3v3h1v2h1v1h1v1h8v-1h1v-1h1v-2h1V8h-1V5h-1V4h-1V3H6Z\" fill=\"#fff\" fill-opacity=\".1\"/>\n<g>\n").concat((_components$beard$val = (_components$beard = components.beard) === null || _components$beard === void 0 ? void 0 : _components$beard.value(components, colors)) !== null && _components$beard$val !== void 0 ? _components$beard$val : '', "\n</g>\n<g>\n").concat((_components$eyes$valu = (_components$eyes = components.eyes) === null || _components$eyes === void 0 ? void 0 : _components$eyes.value(components, colors)) !== null && _components$eyes$valu !== void 0 ? _components$eyes$valu : '', "\n</g>\n<g>\n").concat((_components$eyebrows$ = (_components$eyebrows = components.eyebrows) === null || _components$eyebrows === void 0 ? void 0 : _components$eyebrows.value(components, colors)) !== null && _components$eyebrows$ !== void 0 ? _components$eyebrows$ : '', "\n</g>\n<g>\n").concat((_components$mouth$val = (_components$mouth = components.mouth) === null || _components$mouth === void 0 ? void 0 : _components$mouth.value(components, colors)) !== null && _components$mouth$val !== void 0 ? _components$mouth$val : '', "\n</g>\n<g>\n").concat((_components$hair$valu = (_components$hair = components.hair) === null || _components$hair === void 0 ? void 0 : _components$hair.value(components, colors)) !== null && _components$hair$valu !== void 0 ? _components$hair$valu : '', "\n</g>\n<g>\n").concat((_components$accessori = (_components$accessori2 = components.accessories) === null || _components$accessori2 === void 0 ? void 0 : _components$accessori2.value(components, colors)) !== null && _components$accessori !== void 0 ? _components$accessori : '', "\n</g>\n<g>\n").concat((_components$glasses$v = (_components$glasses = components.glasses) === null || _components$glasses === void 0 ? void 0 : _components$glasses.value(components, colors)) !== null && _components$glasses$v !== void 0 ? _components$glasses$v : '', "\n</g>\n<g>\n").concat((_components$hat$value = (_components$hat = components.hat) === null || _components$hat === void 0 ? void 0 : _components$hat.value(components, colors)) !== null && _components$hat$value !== void 0 ? _components$hat$value : '', "\n</g>\n<g>\n").concat((_components$clothing$ = (_components$clothing = components.clothing) === null || _components$clothing === void 0 ? void 0 : _components$clothing.value(components, colors)) !== null && _components$clothing$ !== void 0 ? _components$clothing$ : '', "\n</g>\n")
    };
  }
};

/*!
 * Pixel Art (@dicebear/pixel-art)
 *
 * Code licensed under MIT License.
 * Copyright (c) 2021 Florian Körner
 *
 * Design "8biticon" by Plastic Jam licensed under MIT.
 * License: https://github.com/dicebear/dicebear/blob/main/packages/%40dicebear/pixel-art/LICENSE
 */
let {
  create,
  meta,
  schema
} = style;
/** @deprecated will be removed in Version 5.0 */

var index = utils.style.createLegacyWrapper(style);

export { create, index as default, meta, schema };
